Library/Homebrew/test/testing_env.rb

```diff
@@ -20,6 +20,25 @@
   abort "Run `bundle install` or install the mocha and minitest gems before running the tests"
 end
 
+if ENV["HOMEBREW_TESTS_COVERAGE"]
+  # Ensure all Ruby files are `require`d for an accurate coverage measurement
+  Dir[File.expand_path("../../**/*.rb", __FILE__)].each do |f|
+    excluded_core_files = [
+      # scripts
+      "build.rb", "postinstall.rb", "test.rb",
+      # mocked files
+      "config.rb", "global.rb",
+    ]
+
+    next if excluded_core_files.any? { |file| f.end_with?("Homebrew/#{file}") }
+
+    # ignore test files
+    next if f.include?("/test/")
+
+    require f
+  end
+end
```

Comment:
> Becuase this feels hack to me and likely to cause trouble: e.g. There may be conflicts, the order of files during require matters.

If this is the case, *this* feels like a hack to me. Correct me if I'm wrong, because my experience with Ruby is still limited, but we should fix those cases by renaming identifiers and fixing the `require`s at the top of those files instead of relying on order and only a subset being `require`d.

Is there an easy way to dump all the modules/classes/methods known to Ruby at a certain point during the program runtime? That would help to systematically check for some of those issues.

> If it doesn't missed out any coverage.

There are a lot of files we don't test yet. After this change, our coverage is down by about 10%. Not super nice, but at least the new number is much closer to reality.